### PR TITLE
unityhub: add a .desktop item

### PR DESCRIPTION
**Pull request by tesq0**
###### Motivation for this change

#124967

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/unityhub/default.nix [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 let
   version = "2.3.2";
-in appimageTools.wrapType2 rec {
+  src = fetchurl {
+    # mirror of https://public-cdn.cloud.unity3d.com/hub/prod/UnityHub.AppImage
+    url = "https://archive.org/download/unity-hub-${version}/UnityHub.AppImage";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/unityhub/default.nix:7]: Why not use web.archive.org?

File: pkgs/development/tools/unityhub/default.nix [hunk 2]
```diff
@@ -16,16 +22,20 @@ in appimageTools.wrapType2 rec {
     libselinux pciutils libpulseaudio libxml2 icu clang cacert
   ]);
 
+  extraInstallCommands =
+    let appimageContents = appimageTools.extractType2 { inherit name src; }; in
+    ''
+      install -Dm444 ${appimageContents}/unityhub.desktop -t $out/share/applications
+      substituteInPlace $out/share/applications/unityhub.desktop \
+      --replace 'Exec=AppRun' 'Exec=${name}'
```
Comment (SuperSandro2000) [inline pkgs/development/tools/unityhub/default.nix:30]:
```suggestion
        --replace 'Exec=AppRun' 'Exec=${name}'
```

File: pkgs/development/tools/unityhub/default.nix [hunk 3]
```diff
@@ -16,16 +22,20 @@ in appimageTools.wrapType2 rec {
...
+      install -m 444 -D ${appimageContents}/unityhub.png \
+      $out/share/icons/hicolor/64x64/apps/unityhub.png
```
Comment (SuperSandro2000) [inline pkgs/development/tools/unityhub/default.nix:32]:
```suggestion
        $out/share/icons/hicolor/64x64/apps/unityhub.png
```

File: pkgs/development/tools/unityhub/default.nix (hunk 1 above)
Comment (tesq0) [reply pkgs/development/tools/unityhub/default.nix:7]: Isn't archive.org the same thing?

Comment (SuperSandro2000) [reply pkgs/development/tools/unityhub/default.nix:7]: No, web.archive.org can very easily archive web urls and updating them is super easy. To get something on archive.org you need to manually upload it.

Comment (tesq0) [reply pkgs/development/tools/unityhub/default.nix:7]: That's a good point. I made a separate issue (#132781) for it as it's not related to the PR (src.url was not changed here, just moved up to a let expr)

**Pull request merged**